decidim-comments/app/cells/decidim/comments/comments_cell.rb

```diff
@@ -4,13 +4,11 @@ module Decidim
   module Comments
     # A cell to display a comments section for a commentable object.
     class CommentsCell < Decidim::ViewModel
+      include UserRoleChecker
       delegate :user_signed_in?, to: :controller
-      def add_comment
-        return if single_comment?
-        return if comments_blocked?
-        return if user_comments_blocked?
 
-        render :add_comment
+      def add_comment
+        render :add_comment if show_comments?
```

Comment:
```suggestion
        render :add_comment if can_add_comments?
```
Maybe something like this is more semantinc. Because this is for showing the box that allows to comment right?

Reply:
I see, thanks.